src/schema/models/schemaManager.ts

```diff
@@ -19,12 +23,31 @@ export class SchemaManager {
     return this.schemas.find((schema) => schema.name === name);
   }
 
-  public map(name: string, tags: Tags): Tags {
-    return Object.entries(tags).reduce((acc, [key, value]) => {
+  public async map(name: string, tags: Tags): Promise<Tags> {
+    const finalTagsObj: Tags = {};
+    const redisKeysArr: string[] = [];
+    const domainFields = await this.domainFieldsRepo.getDomainFieldsList();
+
+    Object.entries(tags).forEach(([key, value]) => {
       if (this.keyIgnoreSets[name].has(key)) {
-        return acc;
+        return;
+      }
+      if (domainFields.has(key.toUpperCase())) {
+        if (value !== null) {
+          redisKeysArr.push(`${key.toUpperCase()}:${value.toString()}`);
+        }
       }
-      return { ...acc, [`${name}_${key}`]: value };
-    }, {});
+      finalTagsObj[`${name}_${key}`] = value;
```

Comment:
i think we checked, and its faster to create a new object than adding another key to object.

Reply:
done